diff -purN nxcomp.orig/Makefile.in nxcomp/Makefile.in
--- nxcomp.orig/Makefile.in	2006-06-19 12:57:55.000000000 -0500
+++ nxcomp/Makefile.in	2006-07-08 19:40:30.000000000 -0500
@@ -37,7 +37,7 @@ LIBVERSION=@LIBVERSION@
 
 CXX         = @CXX@
 CXXFLAGS    = @CXXFLAGS@ @X_CFLAGS@ @DEFS@ \
-              -Wall -Wpointer-arith
+              -Wall -Wpointer-arith -fPIC
 CXXINCLUDES =
 CXXDEFINES  =
 
@@ -47,7 +47,7 @@ CXXDEFINES  =
 
 CC          = @CC@
 CCFLAGS     = @CFLAGS@ @X_CFLAGS@ @DEFS@ \
-              -Wall -Wpointer-arith
+              -Wall -Wpointer-arith -fPIC
 CCINCLUDES  =
 CCDEFINES   =
 
