diff -purN nxcompsh.orig/Makefile.in nxcompsh/Makefile.in
--- nxcompsh.orig/Makefile.in	2006-01-23 11:50:44.000000000 -0600
+++ nxcompsh/Makefile.in	2006-07-10 17:26:49.000000000 -0500
@@ -32,7 +32,7 @@ LIBVERSION=@LIBVERSION@
 
 CXX         = @CXX@
 CXXFLAGS    = @CXXFLAGS@ @X_CFLAGS@ @DEFS@ \
-              -Wall -Wpointer-arith
+              -Wall -Wpointer-arith -fPIC
 CXXINCLUDES =
 CXXDEFINES  =
 
@@ -43,7 +43,7 @@ CXXDEFINES  =
 
 CC          = @CC@
 CCFLAGS     = @X_CFLAGS@ @DEFS@ \
-              -Wall -Wpointer-arith
+              -Wall -Wpointer-arith -fPIC
 CCINCLUDES  =
 CCDEFINES   =
 
@@ -106,7 +106,7 @@ LIBCYGARCHIVE = lib$(LIBRARY).dll.a
 
 PROGRAM = test-nxcompsh
 
-all: depend $(LIBFULL) $(LIBLOAD) $(LIBSHARED) $(LIBARCHIVE) $(LIBCYGSHARED) $(LIBCYGARCHIVE) 
+all: depend $(LIBFULL) $(LIBLOAD) $(LIBSHARED) $(LIBARCHIVE) 
 
 MSRC   = Main.cpp
 
@@ -178,7 +178,7 @@ install.man:
 
 clean:
 		-rm -f *~ *.o *.bak st?????? core core.* *.out.* \
-		$(LIBFULL) $(LIBLOAD) $(LIBSHARED) $(LIBARCHIVE) $(PROGRAM) $(PROGRAM).exe $(LIBCYGSHARED) $(LIBCYGARCHIVE)
+		$(LIBFULL) $(LIBLOAD) $(LIBSHARED) $(LIBARCHIVE) $(PROGRAM) $(PROGRAM).exe
 
 distclean:	clean
 		-rm -f config.status config.log config.cache depend.status Makefile tags
