diff -purN nxservice.orig/Makefile.in nxservice/Makefile.in
--- nxservice.orig/Makefile.in	2006-03-30 08:23:34.000000000 -0600
+++ nxservice/Makefile.in	2006-07-10 18:56:36.000000000 -0500
@@ -12,7 +12,7 @@ LIBVERSION=@LIBVERSION@
 #
 CXX      = @CXX@
 CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@ @DEFS@ \
-           -Wall -Wpointer-arith 
+           -Wall -Wpointer-arith -fPIC 
 
 CXXINCLUDES = -I. -I../nxcomp -I../nxcompsh
 
