diff -u -x '*.kde*' -r x2gogroupadministration-2.0.1/src/x2gogroupadministration.cpp x2gogroupadministration-2.0.1-mod/src/x2gogroupadministration.cpp
--- x2gogroupadministration-2.0.1/src/x2gogroupadministration.cpp	2009-02-09 14:39:30.000000000 +0100
+++ x2gogroupadministration-2.0.1-mod/src/x2gogroupadministration.cpp	2009-05-12 19:10:45.000000000 +0200
@@ -88,6 +88,8 @@
         QTextStream stream( &file );
         uri = stream.readLine();
         base = stream.readLine();
+        bindDN = stream.readLine();
+        bindDNPw = stream.readLine();
         file.close();
     }
     else
@@ -101,6 +103,10 @@
     base=lst[1];
     lst=lst.split("//",uri);    
     uri=lst[1];
+    lst=lst.split("\"",bindDN);
+    bindDN=lst[1];
+    lst=lst.split("\"",bindDNPw);
+    bindDNPw=lst[1];
     kdDebug()<<"base: "<<base<<" uri: "<<uri<<endl;
     QVBoxLayout* lay=new QVBoxLayout(this);
     QFrame* mainFrame=new QFrame(this);
@@ -243,17 +249,17 @@
     mainLay->addWidget(gbGroupInfo);
     mainLay->setSpacing(10);
     mainLay->setMargin(7);
-    QString passfile="/etc/libnss-ldap.secret";
-    QFile file1(passfile);
-    QCString tpass;
-    if (file1.open(IO_ReadOnly))
-    {
-        QTextStream in(&file1);
-        in>>tpass;
-        file1.close();
-    }
+//     QString passfile="/etc/libnss-ldap.secret";
+//     QFile file1(passfile);
+//     QCString tpass;
+//     if (file1.open(IO_ReadOnly))
+//     {
+//         QTextStream in(&file1);
+//         in>>tpass;
+//         file1.close();
+//     }
     kdDebug()<<"Init ldap"<<endl;
-    ld=new LDAPSession(uri,389,"cn=ldapadmin,"+base,tpass.data(),false,true);
+    ld=new LDAPSession(uri,389,(const char*)bindDN,(const char*)bindDNPw,true,false);
     kdDebug()<<"ok"<<endl;
     gbGroupInfo->setEnabled(false);
     connect(lvUsers,SIGNAL(expanded(QListViewItem*)),
Nur in x2gogroupadministration-2.0.1-mod/src: x2gogroupadministration.cpp~.
diff -u -x '*.kde*' -r x2gogroupadministration-2.0.1/src/x2gogroupadministration.h x2gogroupadministration-2.0.1-mod/src/x2gogroupadministration.h
--- x2gogroupadministration-2.0.1/src/x2gogroupadministration.h	2009-02-09 14:39:30.000000000 +0100
+++ x2gogroupadministration-2.0.1-mod/src/x2gogroupadministration.h	2009-05-12 19:09:08.000000000 +0200
@@ -59,6 +59,8 @@
     KListView* lvUsers;
     QString base;
     QString uri;
+    QString bindDN;
+    QString bindDNPw;
     KLineEdit* leName;
     KLineEdit* leDescription;
     KListBox* lbNativeUsers;
Nur in x2gogroupadministration-2.0.1-mod/src: x2gogroupadministration.h~.
