diff -uBb --recursive common.orig/nxcompext/Makefile.in common/nxcompext/Makefile.in
--- common.orig/nxcompext/Makefile.in	2006-11-22 22:30:10.000000000 +0000
+++ common/nxcompext/Makefile.in	2006-11-22 22:34:08.000000000 +0000
@@ -19,7 +19,7 @@
 
 CC          = @CC@
 CCFLAGS     = $(CXXFLAGS)
-CCINCLUDES  = -I. -I../nxcomp 
+CCINCLUDES  = -I. -I../nxcomp -I../nx-X11/lib/X11 -I../nx-X11/include -I../nx-X11/programs/Xserver/include
 CCDEFINES   =
 
 LDFLAGS     = @LDFLAGS@ -L../nx-X11/lib/X11 -L../nxcomp
