diff -uBb --recursive client.orig/nxdesktop/Makefile client/nxdesktop/Makefile
--- client.orig/nxdesktop/Makefile	2006-11-23 07:49:51.000000000 +0000
+++ client/nxdesktop/Makefile	2006-11-23 07:52:17.000000000 +0000
@@ -30,8 +30,8 @@
 mandir      = $(prefix)/man
 datadir     = $(prefix)/share
 
-INCLUDES = -I../nxcomp -I../nxcompext
-LDLIBS   = -lXpm -L../nxcompext -lXcompext -L../nxcomp -lXcomp -lX11 -lXext
+INCLUDES = -I../../common/nxcomp -I../../common/nxcompext
+LDLIBS   = -L../../common/nx-X11 -lXpm -L../../common/nxcompext -lXcompext -L../../common/nxcomp -lXcomp -lX11 -lXext
 VERSION     = 1.5.0-2
 KEYMAP_PATH = $(datadir)/keymaps-windows/
 
