diff -uBb --recursive server.orig/nxnode/src/configure server/nxnode/src/configure
--- server.orig/nxnode/src/configure	2006-11-23 22:07:58.000000000 +0000
+++ server/nxnode/src/configure	2006-11-23 22:08:43.000000000 +0000
@@ -78,9 +78,7 @@
 fi
 
 echo "Checking perl path"
-out=`../../perl/bin/perl -V:bin`
-eval $out
-perlbin="$bin/perl"
+perlbin="/usr/bin/perl"
 
 echo "Created Makefile"
 cp -f Makefile.in Makefile
