diff -purN freenx-0.5.0.orig/nxclient freenx-0.5.0/nxclient
--- freenx-0.5.0.orig/nxclient	2005-08-02 12:49:08.000000000 -0500
+++ freenx-0.5.0/nxclient	2006-07-11 20:04:32.000000000 -0500
@@ -94,8 +94,8 @@ utility_printer()
 {
 	UTILITY_DRIVERS_CACHE="$HOME/.nx/config/drivers.cache"
 	[ -n "$USER_FAKE_HOME" ] && UTILITY_DRIVERS_CACHE="$USER_FAKE_HOME/.nx/config/drivers.cache"
-	UTILITY_NXPRINT="nxprint"
-	[ -n "$PATH_BIN" ] && UTILITY_NXPRINT="$PATH_BIN/nxprint"
+	UTILITY_NXPRINT="nxprint.freenx"
+	[ -n "$PATH_BIN" ] && UTILITY_NXPRINT="$PATH_BIN/nxprint.freenx"
 	case "$1" in 
 		get)
 			grep "driver|$2|" "$UTILITY_DRIVERS_CACHE" 2>/dev/null
diff -purN freenx-0.5.0.orig/nxnode freenx-0.5.0/nxnode
--- freenx-0.5.0.orig/nxnode	2006-07-01 12:03:39.000000000 -0500
+++ freenx-0.5.0/nxnode	2006-07-11 20:03:13.000000000 -0500
@@ -959,10 +959,10 @@ cmd_node_smbmount()
 	ERROR=$(PASSWD="$password" "$COMMAND_SMBMOUNT" "//$computername/$rdir" "$HOME/$dir" -o username="$username,ip=127.0.0.1,port=$port" 2>&1)
 	if [ $? -eq 0 ]
 	then
-		$PATH_BIN/nxclient -dialog ok -caption "NXServer Message" -message "Info: Share: '//$computername/$rdir' mounted on: '$HOME/$dir'" -noautokill -display :$display &
+		$PATH_BIN/nxclient.freenx -dialog ok -caption "NXServer Message" -message "Info: Share: '//$computername/$rdir' mounted on: '$HOME/$dir'" -noautokill -display :$display &
 		echo "$HOME/$dir" >> "$USER_FAKE_HOME/.nx/C-$SERVER_NAME-$display-$sessionid/scripts/mpoint"
 	else
-		$PATH_BIN/nxclient -dialog ok -caption "NXServer Message" -message "Info: Share: '//$computername/$rdir' failed to mount: $ERROR" -noautokill -display :$display &
+		$PATH_BIN/nxclient.freenx -dialog ok -caption "NXServer Message" -message "Info: Share: '//$computername/$rdir' failed to mount: $ERROR" -noautokill -display :$display &
 	fi
 }
 
@@ -992,7 +992,7 @@ cmd_node_addprinter()
 		DEVICE_URI="ipp://localhost:$port/printers/$printer"
 		NAME="$printer"
 	fi
-	MODEL=$($PATH_BIN/nxclient -printer "$NAME" -noautokill -display :$display)
+	MODEL=$($PATH_BIN/nxclient.freenx -printer "$NAME" -noautokill -display :$display)
 	[ -z "$MODEL" -o "$MODEL" = "cancel: aborted" ] && return
 	
 	PUBLIC="-u allow:$USER"
diff -purN freenx-0.5.0.orig/nxprint freenx-0.5.0/nxprint
--- freenx-0.5.0.orig/nxprint	2005-08-02 12:49:08.000000000 -0500
+++ freenx-0.5.0/nxprint	2006-07-11 20:05:17.000000000 -0500
@@ -30,11 +30,11 @@ done
 
 if [ "$HELP" = "yes" ]
 then
-        echo "nxprint - Prints a list of available drivers or printers"
+        echo "nxprint.freenx - Prints a list of available drivers or printers"
 	echo ""
-        echo "Syntax: nxprint --help"
-        echo "        nxprint --drivers"
-        echo "        nxprint --printes"
+        echo "Syntax: nxprint.freenx --help"
+        echo "        nxprint.freenx --drivers"
+        echo "        nxprint.freenx --printes"
         echo
         echo "  --drivers                prints a list of available CUPS drivers"
         echo "  --printers               prints a list of available CUPS printers"
diff -purN freenx-0.5.0.orig/nxserver freenx-0.5.0/nxserver
--- freenx-0.5.0.orig/nxserver	2006-07-01 12:03:07.000000000 -0500
+++ freenx-0.5.0/nxserver	2006-07-11 20:03:51.000000000 -0500
@@ -90,7 +90,7 @@ passdb_add_user()
 	cp -f $NX_ETC_DIR/passwords $NX_ETC_DIR/passwords.orig
 	echo "$PASSDB_CHUSER:*" >> $NX_ETC_DIR/passwords
 	# deactivated to avoid problems with comm-server
-	su - $PASSDB_CHUSER -c "$PATH_BIN/nxnode --setkey"
+	su - $PASSDB_CHUSER -c "$PATH_BIN/nxnode.freenx --setkey"
 }
 
 passdb_list_user()
@@ -567,7 +567,7 @@ do
 			then
 				log 6 -n "ssh "
 				export COMMAND_SSH			
-				echo "$PASS" | $PATH_BIN/nxnode-login -- ssh "$USER" "$SSHD_PORT" "$PATH_BIN/nxnode" --check 2>&1 >/dev/null
+				echo "$PASS" | $PATH_BIN/nxnode-login -- ssh "$USER" "$SSHD_PORT" "$PATH_BIN/nxnode.freenx" --check 2>&1 >/dev/null
 				if [ $? -eq 0 ]
 				then
 					LOGIN_SUCCESS="1"
@@ -579,7 +579,7 @@ do
 			if [ "$ENABLE_SU_AUTHENTICATION" = "1" -a "$LOGIN_SUCCESS" = "0" ]
 			then
 				log 6 -n "su "
-				echo "$PASS" | $PATH_BIN/nxnode-login -- su "$USER" "" "$PATH_BIN/nxnode" --check 2>&1 >/dev/null
+				echo "$PASS" | $PATH_BIN/nxnode-login -- su "$USER" "" "$PATH_BIN/nxnode.freenx" --check 2>&1 >/dev/null
 				if [ $? -eq 0 ]
 				then
 					LOGIN_SUCCESS="1"
@@ -636,7 +636,7 @@ nxnode_start()
 	:
 	#CMD="$1"
 	#shift
-	#echo "$@" | $PATH_BIN/nxnode "$CMD"
+	#echo "$@" | $PATH_BIN/nxnode.freenx "$CMD"
 }
 
 #NX> 1002 Commit
@@ -666,15 +666,15 @@ server_nxnode_start()
 	if [ "$LOGIN_METHOD" = "SSH" ]
 	then
 	    export COMMAND_SSH
-	    echo "$PASS" | NXNODE_TOSEND="$@" $PATH_BIN/nxnode-login -- ssh "$USER" "$SSHD_PORT" "$PATH_BIN/nxnode" "$CMD" 2>&1 | log_tee
+	    echo "$PASS" | NXNODE_TOSEND="$@" $PATH_BIN/nxnode-login -- ssh "$USER" "$SSHD_PORT" "$PATH_BIN/nxnode.freenx" "$CMD" 2>&1 | log_tee
 	elif [ "$LOGIN_METHOD" = "SU" ]
 	then
-	    echo "$PASS" | NXNODE_TOSEND="$@" $PATH_BIN/nxnode-login -- su "$USER" "" "$PATH_BIN/nxnode" "$CMD" 2>&1 | log_tee
+	    echo "$PASS" | NXNODE_TOSEND="$@" $PATH_BIN/nxnode-login -- su "$USER" "" "$PATH_BIN/nxnode.freenx" "$CMD" 2>&1 | log_tee
 	elif [ "$LOGIN_METHOD" = "USERMODE" ]
 	then
-	    echo "$@" | $PATH_BIN/nxnode "$CMD" 2>&1 | log_tee
+	    echo "$@" | $PATH_BIN/nxnode.freenx "$CMD" 2>&1 | log_tee
 	else
-	    echo "$@" | $COMMAND_SSH -l "$USER" "$NODE_HOSTNAME" -p $SSHD_PORT -x -2 -i $NX_ETC_DIR/users.id_dsa -o 'PubkeyAuthentication yes' -o 'RSAAuthentication yes' -o 'RhostsAuthentication no' -o 'PasswordAuthentication no' -o 'RhostsRSAAuthentication no' -o 'StrictHostKeyChecking no' $PATH_BIN/nxnode "$CMD" | log_tee
+	    echo "$@" | $COMMAND_SSH -l "$USER" "$NODE_HOSTNAME" -p $SSHD_PORT -x -2 -i $NX_ETC_DIR/users.id_dsa -o 'PubkeyAuthentication yes' -o 'RSAAuthentication yes' -o 'RhostsAuthentication no' -o 'PasswordAuthentication no' -o 'RhostsRSAAuthentication no' -o 'StrictHostKeyChecking no' $PATH_BIN/nxnode.freenx "$CMD" | log_tee
 	fi
 }
 
@@ -1488,14 +1488,14 @@ cmd_terminate()
 			--suspend)
 				if [ "$cmd_status" = "Running" ] && stringinstring "unix-" "$cmd_type"
 				then
-					echo "sessionid=$cmd_sessionid" | su - "$cmd_user" -c "$PATH_BIN/nxnode --suspend"
+					echo "sessionid=$cmd_sessionid" | su - "$cmd_user" -c "$PATH_BIN/nxnode.freenx --suspend"
 				fi
 			;;
 			--terminate)
-				echo "sessionid=$cmd_sessionid" | su - "$cmd_user" -c "$PATH_BIN/nxnode --terminate"
+				echo "sessionid=$cmd_sessionid" | su - "$cmd_user" -c "$PATH_BIN/nxnode.freenx --terminate"
 			;;
 			--force-terminate)
-				echo "sessionid=$cmd_sessionid" | su - "$cmd_user" -c "$PATH_BIN/nxnode --terminate"
+				echo "sessionid=$cmd_sessionid" | su - "$cmd_user" -c "$PATH_BIN/nxnode.freenx --terminate"
 				session_close $cmd_sessionid
 			;;
 			esac
@@ -1526,7 +1526,7 @@ cmd_send()
 			# is it a "good" session?
 			if [ "$cmd_status" = "Running" ] && stringinstring "unix-" "$cmd_type"
 			then
-				su - "$cmd_user" -c "$PATH_BIN/nxclient --dialog ok --caption \"NX Administrator Message\" --message \"$@\" --noautokill -display \":$cmd_display\"" &
+				su - "$cmd_user" -c "$PATH_BIN/nxclient.freenx --dialog ok --caption \"NX Administrator Message\" --message \"$@\" --noautokill -display \":$cmd_display\"" &
 				disown $!
 			fi
 	done
@@ -1539,7 +1539,7 @@ cmd_send()
 
 if [ $UID -ne 0 ]
 then
-	[ "$1" = "--agent" ] && exec $PATH_BIN/nxnode "$@"
+	[ "$1" = "--agent" ] && exec $PATH_BIN/nxnode.freenx "$@"
 	[ "$1" != "--passwd" ] && cmd_usage
 	cmd_user_passwd
 	exit 0
