--- freenx-server/nxnode	2005-08-07 12:53:57.000000000 +0300
+++ freenx-server/nxnode	2006-02-02 00:00:25.000000000 +0200
@@ -22,6 +22,26 @@
 
 echo "NX> 1000 NXNODE - Version $NX_VERSION $NX_LICENSE"
 
+### Patch for Xorg-x11 >=7.0 by Oskari Rauta begins here ###
+
+# Meet me at IRC-net/#gentoo as jake1981 or jake___
+# or mail me to jake1981 under domain luukku.com
+# if you got questions. Implement/use or develop it freely.
+
+# Reason of patch:
+# Keyboard doesn't work in Xorg-x11 >= 7.0 if XKEYSYMDB isn't exported
+# and pointed to correct file. Hopefully your file is located in same
+# place as mine (default installation location).
+
+if test `X -version 2>&1 | grep "X Window System Version" | awk '{ print $5 }' | awk -F. '{ print $1 }'` -ge 7
+	then
+		export XKEYSYMDB=/usr/share/X11/XKeysymDB
+		echo "NX> NXNODE - Detected Xorg-X11 Version 7.0 or newer."
+		echo "NX> NXNODE - Exporting XKEYSYMDB for enabling keyboard."
+	fi
+
+### Patch for Xorg-x11 >= 7.0 by Oskari Rauta ends here..
+
 if [ "$1" != "--check" -a "$1" != "--setkey" -a "$1" != "--agent" ]
 then 
 	read CMDLINE
