diff -u -x '*.kde*' -r x2gosystemadministration-2.0.1/src/x2gosystemadministration.cpp x2gosystemadministration-2.0.1-mod/src/x2gosystemadministration.cpp
--- x2gosystemadministration-2.0.1/src/x2gosystemadministration.cpp	2009-02-09 14:39:39.000000000 +0100
+++ x2gosystemadministration-2.0.1-mod/src/x2gosystemadministration.cpp	2009-05-12 19:01:36.000000000 +0200
@@ -108,6 +108,8 @@
 		QTextStream stream ( &file );
 		uri = stream.readLine();
 		base = stream.readLine();
+                bindDN = stream.readLine();
+                bindDNPw = stream.readLine();
 		file.close();
 	}
 	else
@@ -120,6 +122,10 @@
 	base=lst[1];
 	lst=lst.split ( "//",uri );
 	uri=lst[1];
+        lst=lst.split("\"", bindDN);
+        bindDN=lst[1];
+        lst=lst.split("\"",bindDNPw);
+        bindDNPw=lst[1];
 
 	QVBoxLayout* lay=new QVBoxLayout ( this );
 	QFrame* mainFrame=new QFrame ( this );
@@ -300,16 +306,16 @@
 	}
 	else
 	{
-		QString passfile="/etc/libnss-ldap.secret";
-		QFile file ( passfile );
-		QCString tpass;
-		if ( file.open ( IO_ReadOnly ) )
-		{
-			QTextStream in ( &file );
-			in>>tpass;
-			file.close();
-		}
-		ld=new LDAPSession ( uri,389,"cn=ldapadmin,"+base,tpass.data(),false,true );
+// 		QString passfile="/etc/libnss-ldap.secret";
+// 		QFile file ( passfile );
+// 		QCString tpass;
+// 		if ( file.open ( IO_ReadOnly ) )
+// 		{
+// 			QTextStream in ( &file );
+// 			in>>tpass;
+// 			file.close();
+// 		}
+		ld=new LDAPSession ( uri,389,(const char*)bindDN,(const char*)bindDNPw,true,false);
 	}
 	load();
 };
Nur in x2gosystemadministration-2.0.1-mod/src: x2gosystemadministration.cpp~.
diff -u -x '*.kde*' -r x2gosystemadministration-2.0.1/src/x2gosystemadministration.h x2gosystemadministration-2.0.1-mod/src/x2gosystemadministration.h
--- x2gosystemadministration-2.0.1/src/x2gosystemadministration.h	2009-02-09 14:39:39.000000000 +0100
+++ x2gosystemadministration-2.0.1-mod/src/x2gosystemadministration.h	2009-05-12 18:59:54.000000000 +0200
@@ -73,6 +73,8 @@
 		QCheckBox* chbNotStartSndSys;
 		QString base;
 		QString uri;
+                QString bindDN;
+                QString bindDNPw;
 		QStringList allGroups;
 		QStringList otherGroups;
 		LDAPSession* ld;
Nur in x2gosystemadministration-2.0.1-mod/src: x2gosystemadministration.h~.
