diff -purN nxnode.orig/nxdesktop/Makefile nxnode/nxdesktop/Makefile
--- nxnode.orig/nxdesktop/Makefile	2006-03-28 16:03:18.000000000 -0600
+++ nxnode/nxdesktop/Makefile	2006-07-30 11:01:29.000000000 -0500
@@ -30,8 +30,8 @@ bindir      = $(exec_prefix)/bin
 mandir      = $(prefix)/man
 datadir     = $(prefix)/share
 
-INCLUDES = -I../nxcomp -I../nxcompext -I../nxesd
-LDLIBS   = -lXpm -L../nxcompext -lXcompext -L../nxcomp -L../nxesd/.libs -lXcomp -lX11 -lXext
+INCLUDES = -I/usr/NX/include -I../nxcompext -I../nxesd
+LDLIBS   = -lXpm -L../nxcompext -lXcompext -L/usr/NX/lib -L../nxesd/.libs -lXcomp -lX11 -lXext
 VERSION     = 1.5.0-2
 KEYMAP_PATH = $(datadir)/keymaps-windows/
 
diff -purN nxnode.orig/nxdesktop/configure nxnode/nxdesktop/configure
--- nxnode.orig/nxdesktop/configure	2006-03-21 13:22:20.000000000 -0600
+++ nxnode/nxdesktop/configure	2006-07-30 11:00:31.000000000 -0500
@@ -391,7 +391,7 @@ rpath=`echo $rpath |sed 's/^://'`
 
 case `uname -s` in
   SunOS)
-    echo "INCLUDES	=  -I../nxcomp -I../nxcompext -I/usr/local/include -I/usr/sfw/include">> Makeconf
+    echo "INCLUDES	=  -I/usr/NX/include -I../nxcompext -I/usr/local/include -I/usr/sfw/include">> Makeconf
     ldflags="$ldflags -L/usr/sfw/lib -lsocket -lnsl -R$rpath"
     cflags="$cflags -DHAVE_DIRENT_H=1 -DHAVE_DECL_DIRFD=0 -DDIR_FD_MEMBER_NAME=dd_fd"
     ;;
