diff -purN nxspool.orig/source/include/popt_common.h nxspool/source/include/popt_common.h
--- nxspool.orig/source/include/popt_common.h	2006-06-16 11:52:49.000000000 -0500
+++ nxspool/source/include/popt_common.h	2006-07-30 07:30:21.000000000 -0500
@@ -21,6 +21,8 @@
 #ifndef _POPT_COMMON_H
 #define _POPT_COMMON_H
 
+#include <popt.h>
+
 /* Common popt structures */
 extern struct poptOption popt_common_samba[];
 extern struct poptOption popt_common_connection[];
