diff -purN nxnode.orig/nxviewer/nxviewer/Imakefile nxnode/nxviewer/nxviewer/Imakefile
--- nxnode.orig/nxviewer/nxviewer/Imakefile	2006-02-22 11:23:57.000000000 -0600
+++ nxnode/nxviewer/nxviewer/Imakefile	2006-07-30 11:23:53.000000000 -0500
@@ -26,8 +26,8 @@ XCOMM seems to cause problems on Digital
 DEFINES = -DMITSHM
 #endif
 
-NX_INC = -I../../nxcomp -I../../nxcompext -I../../nx-X11/exports/include
-NX_LIB = -L../../nxcomp -lXcomp -L../../nxcompext -lXcompext
+NX_INC = -I/usr/NX/include -I../../nxcompext -I../../nx-X11/exports/include
+NX_LIB = -L/usr/NX/lib -lXcomp -L../../nxcompext -lXcompext
 
 #ifdef SunArchitecture
 NX_INC += -I/usr/sfw/include -I/usr/include
