diff -ubB --recursive server.orig/nxviewer/nxviewer/Imakefile server/nxviewer/nxviewer/Imakefile
--- server.orig/nxviewer/nxviewer/Imakefile	2006-11-23 20:23:37.000000000 +0000
+++ server/nxviewer/nxviewer/Imakefile	2006-11-23 20:25:41.000000000 +0000
@@ -15,42 +15,24 @@
 /*                                                                         */
 /***************************************************************************/
 
-#ifdef SunArchitecture
-EXTRA_DEFINES = -D__EXTENSIONS__
-#endif
-
 XCOMM Shared memory support works OK on x86 linux, not tested elsewhere but
 XCOMM seems to cause problems on Digital Unix.
 
 #if defined(i386Architecture) && defined(LinuxArchitecture)
 DEFINES = -DMITSHM
-ZLIB_INC = -I../../zlib-1-2.1
-JPEG_INC = -I../../jpeg-6b
 #endif
 
-NX_INC = -I../../nxcomp -I../../nxcompext -I../../nx-X11/exports/include
-NX_LIB = -L../../nxcomp -lXcomp -L../../nxcompext -lXcompext
-
-#ifdef SunArchitecture
-NX_INC += -I/usr/sfw/include -I/usr/include
-NX_LIB += ../../nx-X11/exports/lib/libXpm.a -L../../nx-X11/exports/lib
-ZLIB_INC = -I/usr/local/include
-JPEG_INC = -I/usr/local/include
-#endif
+NX_INC = -I../../../common/nxcomp -I../../../common/nxcompext -I../../../common/nx-X11/exports/include
+NX_LIB = -L../../../common/nxcomp -lXcomp -L../../../common/nxcompext -lXcompext
 
 
-INCLUDES = -I../include -I. $(ZLIB_INC) $(JPEG_INC) -I/usr/include/X11 $(NX_INC)
+INCLUDES = -I../include -I. $(ZLIB_INC) $(JPEG_INC) -I../../../common/nx-X11/lib/X11 -I../../../common/nx-X11/include -I../../../common/nx-X11/programs/Xserver/include  $(NX_INC)
 VNCAUTH_LIB = ./libvncauth.a
-ZLIB_LIB = -L../../zlib-1.2.1 -lz
-JPEG_LIB = -L../../jpeg-6b/.libs -ljpeg
+ZLIB_LIB = -lz
+JPEG_LIB = -ljpeg
 
-USRLIBDIR = ../../nx-X11/exports/lib
+USRLIBDIR = ../../../common/nx-X11/lib/X11
 
-#ifdef SunArchitecture
-USRLIBDIR = /usr/NX/lib
-SUN_LIB = -lX11 -lXext -L/usr/openwin/lib
-JPEG_LIB = ../../jpeg-6b/.libs/libjpeg.a
-#endif
 
 DEPLIBS = XawClientDepLibs $(VNCAUTH_LIB)
 
